tests/integration/static_ip_address_test.yaml

```diff
@@ -0,0 +1,463 @@
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+    address: []
+description: Configure the eth2 ethernet device with the address 2001:db8:1::1/64, dhcp6 disabled and autoconf disabled
+interfaces:
+- name: eth2
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+    address:
+    - ip: 2001:db8:1::1
+      prefix-length: 64
+description: Configure the eth1 ethernet device with IPv6 link local address only
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+description: Configure the eth1 ethernet device with the address 2001:db8:1::1/64, dhcp6 disabled, autoconf disabled
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+    address:
+    - ip: 2001:db8:1::1
+      prefix-length: 64
+description: Configure the eth1 ethernet device with IPv6 disabled
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: false
+description: Configure the eth1 ethernet device with the address 2001:db8:1::1/64, dhcp6 disabled, autoconf disabled
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+    address:
+    - ip: 2001:db8:1::1
+      prefix-length: 64
+description: Configure the eth1 ethernet device with the address 192.0.2.251/24 and dhcp4 disabled
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: false
+    address:
+    - ip: 192.0.2.251
+      prefix-length: 24
+description: Configure the eth1 ethernet device with address 192.0.2.252/30
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    address:
+    - ip: 192.0.2.252
+      prefix-length: 30
+description: Configure the ethernet device eth1 with the address 2001:db8:2::1/64
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    address:
+    - ip: 2001:db8:2::1
+      prefix-length: 64
+description: Configure the eth1 ethernet device with the address 2001:db8:1::1/64, dhcp6 disabled, autoconf disabled
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+    address:
+    - ip: 2001:db8:1::1
+      prefix-length: 64
+description: Configure the ethernet device eth1 with address 192.0.2.251/24 with dhcp4 disabled
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: false
+    address:
+    - ip: 192.0.2.251
+      prefix-length: 24
+description: Configure the ethernet device eth1 based on the MAC address matching with address 192.0.2.251/24 and with profile name test0
+interfaces:
+- name: eth1
+  profile-name: test0
+  type: ethernet
+  state: up
+  identifier: mac-address
+  mac-address: 5A:DF:6D:A1:32:F8
```

Comment:
You missed the MAC address `5A:DF:6D:A1:32:F8`.

`Set interface holding MAC address 5A:DF:6D:A1:32:F8 with 192.0.2.251/24 and profile name test0`

Reply:
Thanks for the catch, I updated it.